source/slang/check.cpp

```diff
@@ -10763,6 +10763,41 @@ static bool doesParameterMatch(
         Slang::_specializeExistentialTypeParams(getLinkage(), m_globalExistentialSlots, args, sink);
     }
 
+    Type* Linkage::specializeType(
+        Type*           unspecializedType,
+        Int             argCount,
```

Comment:
I guess there might be an argument for using Index instead of Int for such scenarios.

Reply:
I can't bring myself to use `Index` for something that is a count. :)
Otherwise, yes, that would make sense.